django/src/rdwatch_scoring/tasks/__init__.py

```diff
@@ -364,17 +390,22 @@ def get_siteobservations_images(
 
 
 @shared_task
-def cancel_generate_images_task(model_run_id: UUID4) -> None:
-    sites = Site.objects.filter(evaluation_run_uuid=model_run_id)
+def cancel_generate_images_task(model_run_uuid: UUID4, proposal: bool) -> None:
+    if proposal:
+        sites = AnnotationProposalSite.objects.filter(
+            annotation_proposal_set_uuid=model_run_uuid
+        )
+    else:
+        sites = Site.objects.filter(evaluation_run_uuid=model_run_uuid)
```

Comment:
Same question here about removing the `proposal` arg to avoid changing the function signature.

```suggestion
def cancel_generate_images_task(model_run_uuid: UUID4) -> None:
    sites = AnnotationProposalSite.objects.filter(
        annotation_proposal_set_uuid=model_run_uuid
    )
    if not sites.exists():
        sites = Site.objects.filter(evaluation_run_uuid=model_run_uuid)
```

Reply:
Pushed an update to address this: https://github.com/ResonantGeoData/RD-WATCH/pull/392/commits/5f73568fe561939fd1f5636214470353c6b5bd94